include/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.hpp

```diff
@@ -61,6 +67,62 @@ class LLA
         const LLA& aLLA, const Length& anEllipsoidEquatorialRadius, const Real& anEllipsoidFlattening
     ) const;
 
+    /// @brief                  Calculate the azimuth angles between this LLA coordinate and another LLA coordinate
+    ///
+    /// @param                  [in] aLLA A LLA coordinate
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 Azimuth angle from this LLA coordinate to the other LLA coordinate, and azimuth angle
+    /// from the other LLA coordinate to this LLA coordinate.
+
+    Pair<Angle, Angle> calculateAzimuthTo(
+        const LLA& aLLA, const Length& anEllipsoidEquatorialRadius, const Real& anEllipsoidFlattening
+    ) const;
+
+    /// @brief                  Calculate a point between this LLA coordinate and another LLA coordinate
+    ///
+    /// @param                  [in] aLLA A LLA coordinate
+    /// @param                  [in] aRatio A ratio
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 A point between the two LLA coordinates
+
+    LLA calculateIntermediateTo(
+        const LLA& aLLA,
+        const Real& aRatio,
+        const Length& anEllipsoidEquatorialRadius,
+        const Real& anEllipsoidFlattening
+    ) const;
+
+    /// @brief                  Propagate this LLA coordinate in provided direction and distance
+    ///
+    /// @param                  [in] aDirection An azimuth angle
+    /// @param                  [in] aDistance A distance
+    /// @param                  [in] anEllipsoidEquatorialRadius An ellipsoid equatorial radius
+    /// @param                  [in] anEllipsoidFlattening An ellipsoid flattening
+    /// @return                 Propagated LLA coordinate
+
+    LLA calculateForward(
```

Comment:
For the naming, what about 
`calculateForwardLLA`

Reply:
Same as above.